### PR TITLE
Unpack atomix dependencies

### PR DESCRIPTION
**Pull request by pihme**
## Description

So basically, this PR unpacks any dependency to atomix in the broker and gateway modules. Whenever there was a dependency on Atomix, I looked at which bits of Atomix are actually needed and lifted those out. The purpose is to get a clearer picture of the dependencies between components, and ultimate to untangle the bootstrap sequence.

Review hints:
We do not need to merge this PR. We can also keep it as a separate branch and the whole exercise was just to gain more insight into the dependencies.

However, having gone through the exercise, I quite like this level of granularity of dependencies for a time being. I also think it is a good first step toward better testability and dependency injection through constructor injection.

## Definition of Done

Code changes:
* [X] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [X] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: atomix/core/src/main/java/io/atomix/core/Atomix.java [hunk 1]
```diff
@@ -109,7 +104,11 @@ protected Atomix(
         Executors.newScheduledThreadPool(
             Math.max(Math.min(Runtime.getRuntime().availableProcessors() * 2, 8), 4),
             Threads.namedThreads("atomix-primitive-%d", LOGGER));
-    partitions = buildPartitionService(config, getMembershipService(), getCommunicationService());
+
+    final PartitionGroupConfig<?> partitionGroupConfig = config.getPartitionGroup();
```
Comment (npepinpe) [inline atomix/core/src/main/java/io/atomix/core/Atomix.java:108]: :thought_balloon: : we might want to "merge" the `PartitionGroup` and the `RaftPartitionGroup` (and related classes) into a single interface. We won't support different types of partitions, but having an interface is still beneficial for testing. But maybe some of the RaftPartitionGroup specific properties might be lifted up? Just thinking out loud.

**Review by npepinpe — APPROVED** [review]
:rocket: 

I see what you mean. I think actually it's a useful intermediate step and it'll simplify the upcoming refactoring.

**Comment by pihme** [comment]
bors retry

**Pull request merged**